va_explorer/va_data_management/management/commands/run_coding_algorithms.py

```diff
@@ -1,15 +1,52 @@
+import argparse
+import pandas as pd
+import time
 from django.core.management.base import BaseCommand
-from va_explorer.va_data_management.utils.coding import run_coding_algorithms
+from django.db.models import F
+from va_explorer.va_data_management.utils.coding import run_coding_algorithms, ALGORITHM_SETTINGS, validate_algorithm_settings
+from va_explorer.va_data_management.models import VerbalAutopsy, CauseOfDeath
 
 
 class Command(BaseCommand):
 
     help = 'Run cause coding algorithms'
 
-    def handle(self, *args, **options):
-        results = run_coding_algorithms()
-        num_coded = len(results['causes'])
-        num_total = len(results['verbal_autopsies'])
-        num_issues = len(results['issues'])
+    def add_arguments(self, parser):
+        # whether or not to overwrite existing CODs - if True, will clear all CODs before running
+        parser.add_argument('--overwrite', type=bool, nargs='?', default=False)
+        parser.add_argument('--cod_fname', type=str, nargs='?', default='old_cod_mapping.csv')
 
-        self.stdout.write(f'Coded {num_coded} verbal autopsies (out of {num_total}) [{num_issues} issues]')
+    def handle(self, **options):
+        ti = time.time()
+        # validate algorithm settings first. Only proceed if settings are valid. 
+        settings_valid = validate_algorithm_settings(ALGORITHM_SETTINGS)
+        if settings_valid:
+            if options['overwrite']:
+                self.clear_and_save_old_cods(options['cod_fname'])
+
+            print('coding all eligible VAs... ')
+            results = run_coding_algorithms()
+            num_coded = len(results['causes'])
+            num_total = len(results['verbal_autopsies'])
+            num_issues = len(results['issues'])
+            self.stdout.write(f'DONE. Total time: {time.time() - ti} secs')
+            self.stdout.write(f'Coded {num_coded} verbal autopsies (out of {num_total}) [{num_issues} issues]')
+        else:
+            raise ValueError(f"At least one invalid algorithm setting in: \n {ALGORITHM_SETTINGS}. See va_data_management.utils.coding.py for valid setting values." )
+
+    def clear_and_save_old_cods(self, cod_fname=None):
+        
+        print('clearing old CODs...')
+        if not cod_fname:
```

Comment:
Don't need to do this check because it will always be either the custom file name or 'old_cod_mapping.csv' because it's the default in `parser.add_argument()`